pkg/faketikv/task.go

```diff
@@ -315,3 +337,28 @@ func (a *addLearner) RegionID() uint64 {
 func (a *addLearner) IsFinished() bool {
 	return a.finished
 }
+
+func processSnapshot(n *Node, stat *snapshotStat, snapshotSize int64) bool {
+	// If the statement is true, it will start to send or receive the snapshot.
+	if stat.remainSize == snapshotSize {
+		if stat.kind == "sending" {
+			n.stats.SendingSnapCount++
+		} else {
+			n.stats.ReceivingSnapCount++
+		}
+	}
+	stat.remainSize -= n.ioRate
+	// The sending or receiving process has not finished yet.
+	if stat.remainSize > 0 {
+		return false
+	}
+	if !stat.finished {
```

Comment:
I think `finished` is redundant as it always equals to `remainSize <= 0`.

Reply:
The reason that I use `finished` here is to make subtracting one from `ReceivingSnapCount` or `SendingSnapCount` happen only once.